rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1007,38 +1088,28 @@ module RustDataFlow implements InputSig<Location> {
   }
 
   /** Holds if path `p` resolves to struct `s`. */
-  private predicate pathResolveToStructCanonicalPath(PathAstNode p, StructCanonicalPath s) {
-    exists(CrateOriginOption crate, string path |
-      resolveExtendedCanonicalPath(p, crate, path) and
-      s = MkStructCanonicalPath(crate, path)
-    )
+  private predicate pathResolveToStruct(PathAstNode p, Struct s) {
```

Comment:
With this change `pathResolveToStruct` and `pathResolveToVariant` are both very simple and identical in implementation. Would it make sense to drop them and inline them or to combine them into one generalized predicate?